From: Arnd Bergmann <arnd@arndb.de>
Subject: zsmalloc: avoid unused-function warning
Date: Tue, 17 Jan 2023 18:04:59 +0100

obj_allocated() can be called from two places that are each inside of an
#ifdef.  When both are disabled, the compiler warns:

mm/zsmalloc.c:900:13: error: 'obj_allocated' defined but not used [-Werror=unused-function]

Rather than trying to figure out the correct #ifdef, mark the trivial
function as 'inline', which implies __maybe_unused and shuts up the
warning.

Link: https://lkml.kernel.org/r/20230117170507.2651972-1-arnd@kernel.org
Fixes: 796c71ac728e ("zsmalloc: fix a race with deferred_handles storing")
Signed-off-by: Arnd Bergmann <arnd@arndb.de>
Reviewed-by: Sergey Senozhatsky <senozhatsky@chromium.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/mm/zsmalloc.c~zsmalloc-fix-a-race-with-deferred_handles-storing-fix
+++ a/mm/zsmalloc.c
@@ -937,7 +937,7 @@ static bool obj_tagged(struct page *page
 	return true;
 }
 
-static bool obj_allocated(struct page *page, void *obj, unsigned long *phandle)
+static inline bool obj_allocated(struct page *page, void *obj, unsigned long *phandle)
 {
 	return obj_tagged(page, obj, phandle, OBJ_ALLOCATED_TAG);
 }
_
